service/history/tasks/task.go

```diff
@@ -38,6 +38,8 @@ type (
 
 	Keys []Key
 
+	Category int32
```

Comment:
- This type has to be defined in `tasks` package instead of the new `queues` package. Otherwise, there'll be cycle dependency problem.
- Renamed the type name to `Category` as `tasks.Category` makes more sense to me than `tasks.QueueType`
- This `Category` type can be regarded as our own history definition for `enumspb.TaskCategory`. So all history service code won't depend on proto generated code. It also allow us to freely add more categories.